content/client-lib-development-guide/features.textile

```diff
@@ -299,7 +299,9 @@ h3(#realtime-connection). Connection
 ** @(RTN15b)@ In order for a connection to be resumed and connection state to be recovered, the client library reconnects to the websocket endpoint with two additional querystring params:
 *** @(RTN15b1)@ @resume@ is the private connection key assigned to the connection when the first @CONNECTED@ @ProtocolMessage@ was received
 *** @(RTN15b2)@ @connection_serial@ is the most recent @ProtocolMessage#connectionSerial@ received from Ably or @Connection#serial@ which should be identical
-** @(RTN15c)@ When a new connection is made using the @resume@ and @connection_serial@ params, the client library must consider the resume to have failed if the @CONNECTED@ @ProtocolMessage@ contains an error or if the @Connection#id@ has changed. When a @Connection@ resume has failed, all channels should be detach with a suitable error reason
+** @(RTN15c)@ The system's response to a resume request can be a @CONNECTED@ message with an error. If the @CONNECTED@ message @connectionId@ matches the prior @connectionId@, then Ably has confirmed to the client that the resume has been successful, the connection state is intact, and all attached channels remain attached. However, if the @connectionId@ has changed and optionally an error is present, then there may have been a non-fatal error (such as the system is unable to send the backlog in full because the limit of backlog messages has been exceeded), at which point the client library must assume that all connection state is lost and a new connection has been initialised
```

Comment:
@paddybyers predictability. We have often discussed that PubNub's claim that messages are in order (mostly) and are resumed (mostly) is a real fault of their product.  This now feels like we are more like them and don't provide guarantees.  I would rather a user knows messages are missed and does a history query knowing the limitations of that, than resumes and has some messages that are lost. In my mind, resume is binary, it either works or it doesn't.